activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java

```diff
@@ -0,0 +1,59 @@
+package org.activiti.cloud.services.audit.jpa.converters;
+
+import org.activiti.api.process.model.events.BPMNTimerEvent;
+import org.activiti.cloud.api.model.shared.events.CloudRuntimeEvent;
+import org.activiti.cloud.api.model.shared.impl.events.CloudRuntimeEventImpl;
+import org.activiti.cloud.api.process.model.events.CloudBPMNTimerCancelledEvent;
+import org.activiti.cloud.api.process.model.impl.events.CloudBPMNTimerCancelledEventImpl;
+import org.activiti.cloud.services.audit.jpa.events.AuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.events.TimerCancelledAuditEventEntity;
+
+public class TimerCancelledEventConverter extends BaseEventToEntityConverter {
+
+    public TimerCancelledEventConverter(EventContextInfoAppender eventContextInfoAppender) {
+        super(eventContextInfoAppender);
+    }
+
+    @Override
+    public String getSupportedEvent() {
+        return BPMNTimerEvent.TimerEvents.TIMER_CANCELLED.name();
+    }
+
+    @Override
+    protected TimerCancelledAuditEventEntity createEventEntity(CloudRuntimeEvent cloudRuntimeEvent) {
+        CloudBPMNTimerCancelledEvent cloudEvent = (CloudBPMNTimerCancelledEvent) cloudRuntimeEvent;
+        return new TimerCancelledAuditEventEntity(cloudEvent.getId(),
```

Comment:
Uncle Bob in the Clean Code book explains that an ideal function is a niladic function (0 arguments)
1 argument monadic function Not ideal
2 arguments dyadic function Not ideal
3 arguments triadic function
More then  3 arguments polyadic function No to be used

Would not be better define an interface as an input parameter and let all the object implement that interface, and do the assignment job only in the super object?

Reply:
Totally agree with Eugenio. All the events are using big constructors, but we need to stop doing so.

Reply:
I do not like big constructors as well... But I have tried to make everything close / similar to already implemented EventEntities... It means that we have to change all, in principle... Should I apply changes for all eventEntities here, or better to make a new PR and apply changes in one shot?

Reply:
Now we fixed that for timers, but you're right, we need to fix that for other events too. Let's do that in a separated PR.